src/ngraph/op/batch_norm.cpp

```diff
@@ -156,63 +114,59 @@ ngraph::op::BatchNormTrainingBackprop::BatchNormTrainingBackprop(
     std::shared_ptr<ngraph::Node> delta)
     : Op("BatchNormTrainingBackprop",
          check_single_output_args({gamma, beta, input, mean, variance, delta}))
-    , epsilon(eps)
+    , m_epsilon(eps)
 
 {
     constructor_validate_and_infer_types();
 }
 
 void ngraph::op::BatchNormTrainingBackprop::validate_and_infer_types()
 {
-    if (validate_punt_if_dynamic())
-    {
-        return;
-    }
-
-    set_output_size(3);
+    PartialShape input_and_delta_shape{get_input_partial_shape(INPUT)};
 
-    NODE_VALIDATION_ASSERT(this, get_input_shape(INPUT).size() == 4)
```

Comment:
Note that I have removed this restriction. (Does that merit a `Changes API` tag?)

Reply:
Yeah I'm thinking it probably does... sigh.